lib/LaTeXML/Package.pm

```diff
@@ -1816,10 +1816,18 @@ sub DefEnvironmentI {
   my $mode = $options{mode};
   $name      = ToString($name)                    if ref $name;
   $paramlist = parseParameters($paramlist, $name) if defined $paramlist && !ref $paramlist;
+  # Magic form: CS with name \begin{env} bypasses some LaTeX
+  # However, in pure LaTeX would usually have expanded to \env
+  # and would have skipped spaces before parsing args, if any.
+  my $paramlist_skips;
+  if ($paramlist && $paramlist->getNumArgs) {
+    $paramlist_skips = LaTeXML::Core::Parameters->new(
+      LaTeXML::Core::Parameter->new('SkipSpaces', 'SkipSpaces'),
+      $paramlist->getParameters); }
```

Comment:
Hm, this is a little subtle. So `->getNumArgs` counts all the arguments which take a parametric value. But it wouldn't count a `Match:` parameter, or a `RequireBrace`, or some of the `Skip` cases.

What is the downside to instead *always* skipping the spaces here?

Reply:
It is subtle and a bit heuristic; trying to mimic that LaTeX would have turned `\begin{env}` into `\env` which normally would have skipped spaces, unless they've made other arrangements. If it skips too far, or in the wrong cases, we end up on the next line which throws some environments off (eg. listings).